src/librustc_mir/borrow_check/borrow_set.rs

```diff
@@ -254,15 +254,16 @@ impl<'a, 'gcx, 'tcx> Visitor<'tcx> for GatherBorrows<'a, 'gcx, 'tcx> {
         context: PlaceContext<'tcx>,
         location: Location,
     ) {
-        self.super_place(place, context, location);
+        let neo_place = self.tcx.as_new_place(place);
+        self.super_neoplace(&neo_place, context, location);
 
         // We found a use of some temporary TEMP...
-        if let Place::Local(temp) = place {
+        if let mir::PlaceBase::Local(temp) = &neo_place.base {
```

Comment:
This is not equivalent to the old code. In particular, the old code was checking for a place `x` and **excluding** places like `x.f`, but you code ignores the projections. If we have the `into_tree` I was talking about earlier, you might write this as

```rust
if let Tree::Leaf(mir::PlaceBase::Local(temp)) = neo_place.as_tree() { ... }
```

but then again maybe we just want something like `neo_place.as_leaf()` that returns `Option<PlaceBase>`?

You could also just check that the `elems` slice is empty, but that will be awkward to do in an `if let` without some helper fn.